include/path_tracking_pid/details/derivative.hpp

```diff
@@ -0,0 +1,32 @@
+#pragma once
+
+#include <path_tracking_pid/details/fifo_array.hpp>
+
+namespace path_tracking_pid::details
+{
+/**
+ * @brief discrete time derivative filter
+ */
+class Derivative
```

Comment:
Should we not create also a class for the integral action to make all filter modules generic?.

Reply:
Also a reset to the integral and derivative action can be a good idea after, for instance an emergency stop, the control loop has been opened

Reply:
Good idea. I've created a integral filter with unit tests